infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batchprocessor/StateTransferBatchProcessor.java

```diff
@@ -31,6 +42,17 @@
      */
     CompletableFuture<TransferBatchResponse> transfer(TransferBatchRequest transferBatchRequest);
 
+    default Set<Long> getNonTransferredAddresses(LogUnitClient client, ImmutableList<Long> addressesToTransfer) {
```

Comment:
Is `addressesToTransfer ` always sorted?

Reply:
We have a checkReadRecords method that should take care of that but I agree that there should be a check for that in this method too. I'll also add the javadoc.